### ISSUE TITLE
scss/selector-no-redundant-nesting-selector

### ISSUE THREAD

**Issue by testerez** [kristerkari/stylelint-scss#170]
I just updated to version 2.1.0 and this became an issue:

```scss
.foo {
  &,
  & .bar {
    // some style
  }
}
```

`[stylelint] Unnecessary nesting selector (&) (scss/selector-no-redundant-nesting-selector)`

I think that this should be a valid use case.

**Comment by kristerkari** [comment]
Thanks! Your example does not compile with Sass, do you have an example that both compiles correctly and warns about unnecessary nesting selector?

**Comment by testerez** [comment]
Oups sorry, I fixed the example.

**Comment by kristerkari** [comment]
This looks like a bug as the rule warns about your code example. That code is something that is not covered by our unit tests, so the rule needs a bug fix.

**Comment by kristerkari** [comment]
@testerez This is actually not a bug. You are getting the error because of `&` that is infront of `.bar`.

So these codes are actually equal:

Fails
```scss
.foo {
  &,
  & .bar {
    // some style
  }
}
```

Passes
```scss
.foo {
  &,
  .bar {
    // some style
  }
}
```

**Comment by testerez** [comment]
Oh yes you're right. Thanks!
